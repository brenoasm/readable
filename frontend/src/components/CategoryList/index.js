import React, { Fragment } from 'react';

import CategoryRow from '../CategoryRow';

const CategoryList = ({ categories, activeRoute }) => (
  <Fragment>
    <CategoryRow key="Home" />
    {Array.isArray(categories) && categories.map(category => (
      <CategoryRow
        key={category.name}
        activeRoute={activeRoute}
        category={category} />
    ))}
  </Fragment>
);

export default CategoryList;
