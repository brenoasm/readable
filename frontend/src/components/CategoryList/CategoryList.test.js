/* eslint-disable */

import React from 'react';

import CategoryList from './';

describe('CategoryList', () => {
  let wrapper;

  beforeEach(() => {
    const categories = [
      {
        name: 'react',
        path: 'react'
      },
      {
        name: 'redux',
        path: 'redux'
      }
    ]

    wrapper = shallow(<CategoryList categories={categories} />);
  })

  afterEach(() => {
    wrapper.unmount();
  })

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('renders three children', () => {
    expect(wrapper.children().length).toEqual(3); // Categorias + o Home que Ã© fixo
  })
});
