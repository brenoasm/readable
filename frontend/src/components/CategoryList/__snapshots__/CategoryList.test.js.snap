// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryList matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CategoryList
    categories={
        Array [
            Object {
              "name": "react",
              "path": "react",
            },
            Object {
              "name": "redux",
              "path": "redux",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <CategoryRow
          category={
                    Object {
                              "name": "Home",
                              "path": "/",
                            }
          }
/>,
        Array [
          <CategoryRow
            activeRoute={undefined}
            category={
                        Object {
                                    "name": "react",
                                    "path": "react",
                                  }
            }
/>,
          <CategoryRow
            activeRoute={undefined}
            category={
                        Object {
                                    "name": "redux",
                                    "path": "redux",
                                  }
            }
/>,
        ],
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "Home",
        "nodeType": "function",
        "props": Object {
          "category": Object {
            "name": "Home",
            "path": "/",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "react",
        "nodeType": "function",
        "props": Object {
          "activeRoute": undefined,
          "category": Object {
            "name": "react",
            "path": "react",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "redux",
        "nodeType": "function",
        "props": Object {
          "activeRoute": undefined,
          "category": Object {
            "name": "redux",
            "path": "redux",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <CategoryRow
            category={
                        Object {
                                    "name": "Home",
                                    "path": "/",
                                  }
            }
/>,
          Array [
            <CategoryRow
              activeRoute={undefined}
              category={
                            Object {
                                          "name": "react",
                                          "path": "react",
                                        }
              }
/>,
            <CategoryRow
              activeRoute={undefined}
              category={
                            Object {
                                          "name": "redux",
                                          "path": "redux",
                                        }
              }
/>,
          ],
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "Home",
          "nodeType": "function",
          "props": Object {
            "category": Object {
              "name": "Home",
              "path": "/",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "react",
          "nodeType": "function",
          "props": Object {
            "activeRoute": undefined,
            "category": Object {
              "name": "react",
              "path": "react",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "redux",
          "nodeType": "function",
          "props": Object {
            "activeRoute": undefined,
            "category": Object {
              "name": "redux",
              "path": "redux",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
